packages/cart/src/models/address.ts

```diff
@@ -0,0 +1,82 @@
+import { DAL } from "@medusajs/types"
+import { generateEntityId } from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Entity,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property
+} from "@mikro-orm/core"
+
+
+type OptionalAddressProps = DAL.EntityDateColumns // TODO: To be revisited when more clear
+
+@Entity({ tableName: "cart_address" })
+export default class Address {
+  [OptionalProps]: OptionalAddressProps
+
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @Property({ columnType: "text", nullable: true })
+  customer_id?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  company?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  first_name?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  last_name?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  address_1?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  address_2?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  city?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  country_code?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  province?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  postal_code?: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  phone?: string | null
+
+  @Property({ columnType: "jsonb", nullable: true })
+  metadata?: Record<string, unknown> | null
+
+  @Property({
+    onCreate: () => new Date(),
+    columnType: "timestamptz",
+    defaultRaw: "now()",
+  })
+  created_at: Date
+
+  @Property({
+    onCreate: () => new Date(),
+    onUpdate: () => new Date(),
+    columnType: "timestamptz",
+    defaultRaw: "now()",
+  })
+  updated_at: Date
+
+  @BeforeCreate()
+  onCreate() {
+    this.id = generateEntityId(this.id, "caaddr")
```

Comment:
You mean like this?

```
class Address {
  @PrimaryKey()
  id = generateEntityId(this.id, "caaddr");
  ...
}
```

Reply:
yeah, would that work?

Reply:
I wonder if this will work, if you want to set the ID your self on creation, e.g.:
```
manager.create(Address, { id: "my-custom-id_1234", address_1: "something", ... })
```

This is the case we were initially trying to cover with the `onCreate`, I believe.

Reply:
since if we do pass a value for id the initializer would not be used it would keep the provided id, otherwise fallback to the initializer. I am just wondering it would allow us to not have to do it in both create and init

Reply:
I can give it a shot. Would be nice to be able to remove one of them.

Reply:
If it works, theoretically it could allow to remove both i think

Reply:
Will punt the tests to a follow-up PR that includes the test suites.

Reply:
sounds good